base_layer/chat_ffi/src/callback_handler.rs

```diff
@@ -103,8 +170,14 @@ impl CallbackHandler {
             "Calling ContactStatusChanged callback function for contact {}",
             data.address(),
         );
-        unsafe {
-            (self.callback_contact_status_change)(Box::into_raw(Box::new(data)));
+
+        match ChatFFIContactsLivenessData::try_from(data) {
+            Ok(data) => unsafe {
```

Comment:
I double checked just to be sure, but this is because the value of `self.callback_contact_status_change` is of type `unsafe extern "C" fn(*mut ChatFFIContactsLivenessData);`